src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java

```diff
@@ -23,540 +23,543 @@ of this software and associated documentation files (the "Software"), to deal
+     }
+ 
+ 
+     /**
+      * Called when a build is finalized.
+      * @param run - A Run object representing a particular execution of Job.
+      */
+     @Override
+     public void onFinalized(Run run) {
+         try {
+             // Process only if job in NOT in excluded and is in included
+             if (!DatadogUtilities.isJobTracked(run.getParent().getFullName())) {
+                 return;
+             }
+             logger.fine("Start DatadogBuildListener#onFinalized");
+ 
+             TraceWriter traceWriter = TraceWriterFactory.getTraceWriter();
+             if (traceWriter == null) {
+                 return;
+             }
+ 
+             // Collect Build Data
+             BuildData buildData;
+             try {
+                 buildData = new BuildData(run, null);
+             } catch (InterruptedException e) {
+                 Thread.currentThread().interrupt();
+                 DatadogUtilities.severe(logger, e, "Interrupted while trying to parse finalized build data");
+                 return;
+             } catch (IOException e) {
+                 DatadogUtilities.severe(logger, e, "Failed to parse finalized build data");
+                 return;
+             }
+ 
+             // APM Traces
+             traceWriter.submitBuild(buildData, run);
+             logger.fine("End DatadogBuildListener#onFinalized");
+ 
+         } catch (InterruptedException e) {
+             Thread.currentThread().interrupt();
+             DatadogUtilities.severe(logger, e, "Interrupted while processing build finalization");
+         } catch (Exception e) {
+             DatadogUtilities.severe(logger, e, "Failed to process build finalization");
+         } finally {
+             // Explicit removal of InvisibleActions used to collect Traces when the Run finishes.
+             cleanUpTraceActions(run);
+         }
+     }
+ 
+     @Override
+     public void onDeleted(Run run) {
+         try {
+             // Process only if job is NOT in excluded and is in included
+             if (!DatadogUtilities.isJobTracked(run.getParent().getFullName())) {
+                 return;
+             }
+             logger.fine("Start DatadogBuildListener#onDeleted");
+ 
+             // Get Datadog Client Instance
+             DatadogClient client = getDatadogClient();
+             if (client == null) {
+                 return;
+             }
+ 
+             // Collect Build Data
+             BuildData buildData;
+             try {
+                 buildData = new BuildData(run, null);
+             } catch (InterruptedException e){
+                 Thread.currentThread().interrupt();
+                 DatadogUtilities.severe(logger, e, "Interrupted while trying to parse deleted build data");
+                 return;
+             } catch (IOException e) {
+                 DatadogUtilities.severe(logger, e, "Failed to parse deleted build data");
+                 return;
+             } catch (Exception e) {
```

Comment:
Can you catch all of these exceptions at once?

Reply:
Done.